.github/workflows/publish.yml

```diff
@@ -1,104 +1,32 @@
-              uses: actions/setup-python@v2
-              with:
-                  python-version: '3.10'
-
-            - name: Install pypa/build
-              run: python -m pip install build
-
-            - name: Build a binary wheel and a source tarball
-              run: >-
-                  python -m
-                  build
-                  --sdist
-                  --wheel
-                  --outdir dist/
-            - name: Upload distribution artifact
-              uses: actions/upload-artifact@v2
-              with:
-                  name: release
-                  path: dist/
-
-
-    publish-test:
-
-        name: Build and publish on TestPyPI
-        if: startsWith(github.ref, 'refs/heads/release/')
-        needs: [build]
-        runs-on: ubuntu-latest
-
-        environment:
-            name: Test PyPI
-            url: https://test.pypi.org/project/aiida-cp2k/
-
-        steps:
-            - uses: actions/download-artifact@v2
-              name: Download distribution artifact
-              with:
-                  name: release
-                  path: dist/
-
-            - name: Publish distribution on Test PyPI
-              uses: pypa/gh-action-pypi-publish@release/v1
-              if: startsWith(github.ref, 'refs/heads/release/')
-              with:
-                  user: __token__
-                  password: ${{ secrets.PYPI_API_TOKEN }}
-                  repository_url: https://test.pypi.org/legacy/
-                  skip_existing: true
-
-    publish:
-
-        name: Build and publish on PyPI
-        if: startsWith(github.ref, 'refs/tags')
-        needs: [build]
-        runs-on: ubuntu-latest
-
-        environment:
-            name: PyPI
-            url: https://pypi.org/project/aiida-cp2k/
-
-        steps:
-
-            - uses: actions/download-artifact@v2
-              name: Download distribution artifact
-              with:
-                  name: release
-                  path: dist/
-
-            - uses: softprops/action-gh-release@v0.1.14
-              name: Create release
-              if: startsWith(github.ref, 'refs/tags/v')
-              with:
-                  files: |
-                      dist/*
-                  generate_release_notes: true
-
-            - name: Publish distribution on PyPI
-              uses: pypa/gh-action-pypi-publish@release/v1
-              with:
-                  user: __token__
-                  password: ${{ secrets.PYPI_API_TOKEN }}
+  publish:
+    runs-on: ubuntu-latest
+    if: github.repository == 'aiidateam/aiida-core' && startsWith(github.ref, 'refs/tags/v')
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - uses: softprops/action-gh-release@v0.1.14
+        name: Create release
+        with:
+            generate_release_notes: true
```

Comment:
How does this work? Is this supposed to create the tag automatically? If so, this will never get called though, because the workflow is only run when a tag is pushed.

Reply:
I typically use `bumpver` to create a tag automatically. I will add the tool in the next PR together with a section in the README explaining how to use it.

Reply:
Ok, so the tag will have been created locally, and then the action is just responsible for creating a release entry on Github. However, did I understand correctly that this also pushes the build to PyPI? If so, then you don't want to have `flit publish` in the next step. Because then you are pushing to PyPI twice.

Reply:
> However, did I understand correctly that this also pushes the build to PyPI?

No, publishing on PyPi is typically done as the next step, see for example https://github.com/nanotech-empa/aiida-gaussian/blob/master/.github/workflows/publish.yml